# HG changeset patch
# User Konstantinos Poulios <logari81@googlemail.com>
# Date 1288778942 -3600
#      Wed Nov 03 11:09:02 2010 +0100
# Node ID d9a8a008c116cc1699e156cbd6b8b701c4611dfa
# Parent  ac6a199be45e43fc74311cea4274bf4011058656
Remove deprecated temporary calculation of plotboxaspectratio for the gnuplot backend

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,15 @@
+2010-11-03  Konstantinos Poulios  <logari81@googlemail.com>
+
+	* plot/private/__actual_axis_position__.m: Remove temporary
+	calculation of plotboxratio for the gnuplot backend when
+	dataaspectratiomode, xlimmode and ylimmode are all set to manual.
+	Now this case is handled in src/graphics.cc .
+
 2010-11-03  Konstantinos Poulios  <logari81@googlemail.com>
 
 	* plot/axis.m: Leave zlim unchanged during "axis tight" on 2D plots
 
 2010-11-03  Ben Abbott <bpabbott@mac.com>
 
 	* miscellaneous/getappdata.m, miscellaneous/isappdata.m,
 	miscellaneous/rmappdata.m, miscellaneous/setappdata.m:
diff --git a/scripts/plot/private/__actual_axis_position__.m b/scripts/plot/private/__actual_axis_position__.m
--- a/scripts/plot/private/__actual_axis_position__.m
+++ b/scripts/plot/private/__actual_axis_position__.m
@@ -48,29 +48,18 @@ function pos = __actual_axis_position__ 
       && strcmp (axis_obj.activepositionproperty, "outerposition"))
     pos_in_pixels = axis_obj.outerposition .* fig_position([3, 4, 3, 4]);
   else
     pos_in_pixels = axis_obj.position .* fig_position([3, 4, 3, 4]);
   endif
     
   nd = __calc_dimensions__ (h);
 
-  if (strcmp (axis_obj.plotboxaspectratiomode, "auto")
-      && strcmp (axis_obj.dataaspectratiomode, "manual")
-      && strcmp (axis_obj.xlimmode, "manual")
-      && strcmp (axis_obj.ylimmode, "manual")
-      && (nd == 2 || all (mod (axis_obj.view, 90) == 0)))
-    ## Force plotboxaspectrato to a manual value
-    dx_dy_dz = [diff(axis_obj.xlim), diff(axis_obj.ylim), diff(axis_obj.zlim)];
-    dx_dy_dz = dx_dy_dz / min (dx_dy_dz(dx_dy_dz>0));
-    axis_obj.plotboxaspectratiomode = "manual";
-    axis_obj.plotboxaspectratio = dx_dy_dz ./ axis_obj.dataaspectratio;
-  endif
-
-  if (strcmp (axis_obj.plotboxaspectratiomode, "manual"))
+  if (strcmp (axis_obj.plotboxaspectratiomode, "manual")
+      || strcmp (axis_obj.dataaspectratiomode, "manual"))
     ## When using {rltb}margin, Gnuplot does not handle the specified
     ## aspect ratio properly, so handle it here.
     if (nd == 2 || all (mod (axis_obj.view, 90) == 0))
       aspect_ratio_2d = axis_obj.plotboxaspectratio(1:2);
     else
       ## FIXME -- this works for "axis square", but has not been
       ##          thoroughly tested for other aspect ratios.
       aspect_ratio_2d = [max(axis_obj.plotboxaspectratio(1:2)), ...
